mempool/src/core_mempool/index.rs

```diff
@@ -77,6 +81,7 @@ pub struct OrderedQueueKey {
     pub address: AccountAddress,
     pub sequence_number: u64,
     pub governance_role: GovernanceRole,
+    pub seqno_type: AccountSequenceNumber,
```

Comment:
should we merge these two fields? and leverage an enum to hide the details?

Reply:
Probably a good idea :) I'll update